controller_interface/include/controller_interface/controller_interface_base.hpp

```diff
@@ -115,8 +115,7 @@ class ControllerInterfaceBase : public rclcpp_lifecycle::node_interfaces::Lifecy
   CONTROLLER_INTERFACE_PUBLIC
   virtual return_type init(
     const std::string & controller_name, const std::string & urdf, unsigned int cm_update_rate,
-    const std::string & node_namespace = "",
-    const rclcpp::NodeOptions & node_options = rclcpp::NodeOptions().enable_logger_service(true));
+    const std::string & node_namespace = "");
```

Comment:
@bmagyar @destogl @christophfroehlich 
I need some feedback from you guys. Do you think it make sense to package these arguments into a structure, so that next time we like to add or remove entities we won't be breaking any API.

Moreover, this PR only provides a method for the user to override the node_options of this particular controller. As per the PR: https://github.com/ros-controls/ros2_control/pull/1293, the CM will need to get these node options from the newly added method and modify them with the arguments of the `params-file` and then somehow feed it back to the `init` method to initialize the LifeCycleNode. 

* We can still keep this `node_options` parameter in the `init` method, but we will have to remove the default argument and also mark it as `final` so it won't be overridden 
* We can package them all together into a single class and then mark this `init` method as `final`, then we can do the logic without many issues. 

One of the above should be fine. Any more ideas are welcome.